### DISCUSSION TITLE
Best Practices for Customizing Node Model Paths

### DISCUSSION THREAD

**Discussion by xingren23** [Comfy-Org/ComfyUI#2849] [Ideas]
Here's a suggestion for best practices regarding model paths for your custom nodes:

1. Support configuration of model paths in an extra_model_path.yaml file.
2. Store models downloaded by users in the root directory /comfyui/models/xxxxx.
3. Store models downloaded during runtime in either the root directory /comfyui/models/xxx or within the node itself at /comfyui/custom_nodes/xxxx/models/xxxx.

This approach provides clear guidelines for developers and ensures consistency in managing model paths  

Feel free to discuss and refine this further!

**Comment by alexisrolland** [comment]
Is there any documentation available somewhere for point "1. Support configuration of model paths in an extra_model_path.yaml file."?

**Reply by christian-byrne** [reply]
Do you mean docs for the extra models paths file itself? If so, there is some documentation in [comfyui model sharing and custom model directory configuration](https://docs.comfy.org/installation/comfyui_portable_windows#2-comfyui-model-sharing-and-custom-model-directory-configuration) section of docs.comfy. 

If there's anything that those docs leave out, I can perhaps try to help answer and also update the docs accordingly.
